From 2c60d0bb7a29c98a95c4ebaa9c146818fa01216f Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 31 May 2021 12:21:42 +0300
Subject: [PATCH] [SW-36011] gaudi: security: close spsram on init

Current implementation keeps the SP SRAM open after the secuity init
which might present a security threat.
Add the SP SRAM configuration entry to the common init loop.

Change-Id: I6242d5327382d00e9fb077824d7ac75ed586b7c5
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index dfdf3f0..dbcdade 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1755,7 +1755,7 @@
 	preboot_secure_pcie_rr(PCIE_DYN_RR_SRAM_END, SDRAM_BASE - 1, SRAM_END);
 
 	/* Apply all already known configurations */
-	for (i = SRAM_BGN_TO_VERS; i <= IMG_END_TO_SRAM_END; i++) {
+	for (i = SRAM_BGN_TO_VERS; i <= SPSRAM_IMAGE; i++) {
 		cfg = &preboot_sec_dyn_cfg[i];
 
 		if (cfg->rr >= 0 && cfg->min && cfg->max)
